File: app/src/main/java/com/datadoghealth/cardiodino/core/AppModule.java
package com.datadoghealth.cardiodino.core;

import android.content.Context;


import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Dagger module for dependency injection of global dependencies
 * Created by Elizabeth on 3/23/2015.
 */

@Module(library=true,
        injects={
                AppClass.class,
        })

public class AppModule {
    private final AppClass application;

    AppModule(AppClass application) {this.application=application;}

    @Provides
    @Singleton
    public Context providesApplicationContext() {return application;}



}

File: app/src/main/java/us/elizr/heartrace/core/AppModule.kt
package us.elizr.heartrace.core

import android.app.Application
import android.content.SharedPreferences
import android.preference.PreferenceManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by elizabethrussell on 3/16/18.
 */
@Module
class AppModule(private val application: Application) {

    @Provides
    @Singleton
    internal fun providesApplication(): Application {
        return application
    }

    @Provides
    @Singleton
    internal fun providesSharedPreferences(application: Application): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(application)
    }



}